src/generators.jl

```diff
@@ -1054,7 +1125,25 @@ Output
 """
 function _get_outedges(A::MatrixNetwork{T},i::Int) where T
 
-   (i >= 1 && i <= size(A,1)) || throw(ArgumentError("i must be in {1,...,size(A,1)}"))
-   return A.ci[A.rp[i]:A.rp[i+1]-1],A.vals[A.rp[i]:A.rp[i+1]-1]
+    @boundscheck (i >= 1 && i <= size(A,1)) || throw(ArgumentError("i must be in {1,...,$(size(A,1))}"))
+    return A.ci[A.rp[i]:A.rp[i+1]-1],A.vals[A.rp[i]:A.rp[i+1]-1]
```

Comment:
Do you need to return an array? can you return a generator or a view?

Reply:
They're now returning views, and matrix_to_list_of_list uses `collect` to allocate the arrays needed.